### PR TITLE
Add support for circles

### PR DESCRIPTION
**Pull request by nickvergessen**
1. Install circles: https://github.com/nextcloud/circles/pull/361
2. Create a circle
3. Try to create a conversation with a circle
4. Try to add a circle to an existing conversation

### CODE REVIEW

**Comment by nickvergessen** [comment]
Rebased due to conflict in RoomController (constant for #2502 was added in same place as a member vairable)

**Comment by nickvergessen** [comment]
>  Circles support is coming. you just request them additionally on the endpoint when asking for results:
https://github.com/nextcloud/spreed/pull/2518/files#diff-fced531870aa624bf64ba668bff611c5R58

>  and then a request similar to group conversations is sent, just including `&source=circles in the url
https://github.com/nextcloud/spreed/pull/2518/files#diff-fced531870aa624bf64ba668bff611c5R85

>  in the end we just add the users of the circle to the conversation, like we do with groups
no membership tracking, etc

File: lib/Controller/RoomController.php [hunk 1]
```diff
@@ -659,6 +724,40 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
...
+			try {
+				$circle = Circles::detailsCircle($newParticipant);
+			} catch (\Exception $e) {
+				return new DataResponse([], Http::STATUS_NOT_FOUND);
+			}
+
+			$participants = [];
+			foreach ($circle->getMembers() as $member) {
+				/** @var Member $member */
+				if ($member->getType() === Member::TYPE_USER || $member->getUserId() === '') {
```
Comment (daita) [inline lib/Controller/RoomController.php]: I would add a check on the Level: $member->isLevel(MEMBER::LEVEL_MEMBER) as in current state, non-yet members (invited or requesting for access) will be invited to the room.

File: lib/Controller/RoomController.php [hunk 2]
```diff
@@ -456,6 +467,60 @@ protected function createGroupRoom(string $targetGroupName): DataResponse {
...
+		$room = $this->manager->createGroupRoom($circle->getName());
+		$room->addUsers([
+			'userId' => $currentUser->getUID(),
+			'participantType' => Participant::OWNER,
+		]);
+
+		$participants = [];
+		foreach ($circle->getMembers() as $member) {
+			/** @var Member $member */
+			if ($member->getType() === Member::TYPE_USER || $member->getUserId() === '') {
```
Comment (daita) [inline lib/Controller/RoomController.php]: I would add a check on the Level: $member->isLevel(MEMBER::LEVEL_MEMBER) as in current state, non-yet members (invited or requesting for access) will be invited to the room.
@daita

File: lib/Controller/RoomController.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Controller/RoomController.php]: Invites would be fine I guess. requests is bad of course :D

Comment (daita) [reply lib/Controller/RoomController.php]: in that case, have a look to ->getStatus() that should returns a string related to https://github.com/nextcloud/circles/blob/12934e8c5c886401231fb5808022891824250bc7/lib/Model/BaseMember.php#L41-L46 :]

**Comment by nickvergessen** [comment]
@daita another review?

**Comment by daita** [comment]
Code looks good to me

**Pull request merged**